utils/gnutls_offset.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+set -e
+
+PROJECT_ROOT_DIR=$(pwd)
+GNUTLS_DIR="${PROJECT_ROOT_DIR}/deps/gnutls"
+OUTPUT_DIR="${PROJECT_ROOT_DIR}/kern"
+
+if [[ ! -f "go.mod" ]]; then
+  echo "Run the script from the project root directory"
+  exit 1
+fi
+
+echo "check file exists: ${GNUTLS_DIR}/.git"
+# skip cloning if the header file of the max supported version is already generated
+if [[ ! -f "${GNUTLS_DIR}/.git" ]]; then
+  echo "check directory exists: ${GNUTLS_DIR}"
+  # skip cloning if the gnutls directory already exists
+  if [[ ! -d "${GNUTLS_DIR}" ]]; then
+    echo "git clone gnutls to ${GNUTLS_DIR}"
+    git clone https://github.com/gnutls/gnutls.git ${GNUTLS_DIR}
+  fi
+fi
+
+
+function run() {
+  git fetch --tags
+  cp -f ${PROJECT_ROOT_DIR}/utils/gnutls_offset.c ${GNUTLS_DIR}/offset.c
+  main_version="3.8"
+
+  for ver in $(seq 7 8); do
+    tag="${main_version}.${ver}"
+    underline_tag=$(echo $tag | tr "." "_")
+    header_file="${OUTPUT_DIR}/gnutls_${underline_tag}_kern.c"
+    header_define="GNUTLS_${underline_tag}_KERN_H"
+
+    if [[ -f ${header_file} ]]; then
+      echo "Skip ${header_file}"
+      continue
+    fi
+    echo "git checkout ${tag}"
+    git checkout ${tag}
+    echo "Generating ${header_file}"
+
+    # init
+    ./bootstrap
+    ./configure
+    clang -I gnulib/lib/ -I lib/ -I . offset.c -o offset
```

Comment:
`lib/` is not the correct path, the correct path should be `lib/includes`?

At least it reported an error in my development environment (ubuntu 22.04 arm64).

```shell
In file included from offset.c:8:
./lib/gnutls_int.h:81:10: fatal error: 'gnutls/gnutls.h' file not found
#include <gnutls/gnutls.h>
         ^~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
Maybe my local env is not pure, I will try it in new container.